packages/cactus-test-plugin-htlc-eth-besu/src/test/typescript/integration/plugin-htlc-eth-besu/get-status-endpoint.test.ts

```diff
@@ -0,0 +1,230 @@
+  t.comment(`RESPONSE: ${JSON.stringify(deployOut)}`);
+
+  t.comment(deployOut.transactionReceipt.contractAddress!);
+  const hashTimeLockAddress = deployOut.transactionReceipt
+    .contractAddress as string;
+
+  //Deploy DemoHelpers
+  const deployOutDemo = await connector.deployContract({
+    web3SigningCredential: {
+      ethAccount: firstHighNetWorthAccount,
+      secret: besuKeyPair.privateKey,
+      type: Web3SigningCredentialType.PRIVATEKEYHEX,
+    },
+    bytecode: DemoHelpers.bytecode,
+    gas: dataTest.estimated_gas,
+  });
+  t.ok(deployOutDemo, "deploy DemoHelpers output is truthy OK");
+  const demoHelpersAddress = deployOutDemo.transactionReceipt
+    .contractAddress as string;
+
+  const balance = await web3.eth.getBalance(firstHighNetWorthAccount);
+  t.comment(balance);
+
+  //new Contract
+  const bodyObj: NewContractObj = {
+    contractAddress: hashTimeLockAddress,
+    inputAmount: 10,
+    outputAmount: 0x04,
+    expiration: dataTest.expiration,
+    hashLock: dataTest.hashLock,
+    receiver: dataTest.receiver,
+    outputNetwork: "BTC",
+    outputAddress: "1AcVYm7M3kkJQH28FXAvyBFQzFRL6xPKu8",
+    keychainId: keychainId,
+    connectorId: connectorId,
+    accountRef: accountRef,
+    privateKeyRef: privateKeyRef,
+    credentialType: Web3SigningCredentialType.PRIVATEKEYHEX,
+    gas: dataTest.estimated_gas,
+  };
+  const resp = await api.newContract(bodyObj);
+  t.ok(resp, "response newContact is OK");
+
+  const { callOutput } = await connector.invokeContract({
+    contractAbi: DemoHelpers.abi,
+    contractAddress: demoHelpersAddress as string,
+    invocationType: EthContractInvocationType.CALL,
+    methodName: "getTxId",
+    params: [
+      firstHighNetWorthAccount,
+      dataTest.receiver,
+      10,
+      dataTest.hashLock,
+      dataTest.expiration,
+    ],
+    web3SigningCredential: {
+      ethAccount: firstHighNetWorthAccount,
+      type: Web3SigningCredentialType.PRIVATEKEYHEX,
+      secret: besuKeyPair.privateKey,
+    },
+    gas: dataTest.estimated_gas,
+  });
+  const balance2 = await web3.eth.getBalance(firstHighNetWorthAccount);
+  t.comment(balance2);
+  t.equal(
+    parseInt(balance),
+    parseInt(balance2) - 10,
+    "Balance of account is OK",
+  );
+
+  // Test for 200 valid response test case
+  const res = await api.getStatus(
+    [callOutput as string],
+    keychainId,
+    hashTimeLockAddress,
+    Web3SigningCredentialType.PRIVATEKEYHEX,
+    connectorId,
+    accountRef,
+    privateKeyRef,
+  );
+  t.comment("Getting result");
+  t.equal(res.status, 200);
+
+  //Test for 500 not found test case
+  try {
+    await api.getStatus(
+      ["0xfake5ba7f06a8b01d0596589f73c19069e21c81e5013b91f408165d1bf623d32"],
+      keychainId,
+      hashTimeLockAddress,
+      Web3SigningCredentialType.PRIVATEKEYHEX,
+      connectorId,
+      accountRef,
+      privateKeyRef,
+    );
+  } catch (error) {
+    t.equal(error.response.status, 500, "HTTP response status are equal");
```

Comment:
@AzaharaC Code 500 should be returned for unknown errors. If the server validated the request and found that it had problems (such as something not being found that the request referred to as something in existence or something being in an invalid format) then a 400 or some other 4xx code could better indicate to end users that they need to fix something in their request.
In general 5xx codes mean that the backend is "at fault" meaning that the developer who sent the request has no action need, but in this case they do (which can be expressed in general with some 4xx codes that in general means that the backend thinks that the request sender is at fault)

https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400

Reply:
Done